x-pack/solutions/security/plugins/security_solution/public/plugin.tsx

```diff
@@ -127,10 +144,19 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
       visibleIn: ['globalSearch', 'home', 'kibanaOverview'],
       euiIconType: APP_ICON_SOLUTION,
       mount: async (params) => {
-        const { renderApp, services, store, subPlugins } = await mountDependencies(params);
-        const { getSubPluginRoutesByCapabilities } = await this.lazyHelpersForRoutes();
+        const {
+          getSubPluginRoutesByCapabilities,
+          registerActions,
+          renderApp,
+          services,
+          store,
+          subPlugins,
+        } = await mountDependencies(params);
 
-        await this.registerActions(store, params.history, core, services);
+        if (!this._actionsRegistered) {
+          this._actionsRegistered = true;
+          registerActions(store, params.history, core, services);
```

Comment:
Isn't this function async now? Don't you think we should await?

Reply:
The results of `registerActions` are not used so I did not think an await was needed. I will defer to your team if await is required. Just let me know.

Reply:
This function is async now because you changed that [here](https://github.com/elastic/kibana/pull/214483/files?file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&file-filters%5B%5D=.yml&owned-by%5B%5D=semd&show-deleted-files=true&show-viewed-files=true#diff-4ee78dcd159a4abfb194303aacdefb17fd0698d2f4e08482e2d87e240e90877c)

Does this really need to be async?

Reply:
> Does this really need to be async?

Yes, because `createFilterAction` from `import { createFilterAction } from '@kbn/unified-search-plugin/public';` is now async to avoid including `createFilterAction` in unifiedSearch bundle

Reply:
Added `await` e1bba66ba32283256beccfe863e11b7308d86b87

Reply:
Okay, but this is not awaiting anything unless we also `await` the `registerDiscoverHistogramActions` call inside `registerUIActions`, right?

Reply:
registerUIActions needs to be async so that registerDiscoverHistogramActions can be async. Let me know if you have any further questions

Reply:
@nreese Sure, let me rephrase the question.

Why is the `registerDiscoverHistogramActions()` call not awaited here?

https://github.com/elastic/kibana/blob/e1bba66ba32283256beccfe863e11b7308d86b87/x-pack/solutions/security/plugins/security_solution/public/app/actions/register.ts#L56

Reply:
Thanks for clarifying, added await 98fc7a77837f766a31a44f61f1d7502d738abfc9